modules/shared_memory/shared_memory.c

```diff
@@ -0,0 +1,43 @@
+#include "shared_memory.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ipc.h>
+#include <sys/shm.h>
+
+int createSharedMemory(int numPlayers) {
+  size_t shmSize = sizeof(SharedMemory) + numPlayers * sizeof(Player);
+  int shmid = shmget(IPC_PRIVATE, shmSize, IPC_CREAT | 0666);
+  if (shmid == -1) {
+    fprintf(stderr, "shmget failed: %s\n", strerror(errno));
+    return -1;
+  }
+  return shmid;
+}
+
+SharedMemory *attachSharedMemory(int shmid) {
+  void *shmaddr = shmat(shmid, NULL, 0);
+  if (shmaddr == (void *) -1) {
+    fprintf(stderr, "shmat failed: %s\n", strerror(errno));
+    return NULL;
+  }
+  return (SharedMemory *) shmaddr;
+}
+
+int detachSharedMemory(SharedMemory *shm) {
```

Comment:
I would add something like this tbh as an utility function:

size_t calculateShmSize(int numPlayers) {
    return sizeof(SharedMemory) + numPlayers * sizeof(Player);
}